packages/@n8n/task-runner/src/task-runner.ts

```diff
@@ -328,20 +338,33 @@ export abstract class TaskRunner extends EventEmitter {
 			this.runningTasks.delete(taskId);
 			return;
 		}
+
+		const controller = new AbortController();
+		this.taskCancellations.set(taskId, controller);
+
+		const taskTimeout = setTimeout(() => {
+			if (!task.cancelled) {
+				controller.abort();
+				this.taskCancellations.delete(taskId);
+			}
+		}, this.taskTimeout * 1_000);
+
 		task.settings = settings;
 		task.active = true;
 		try {
-			const data = await this.executeTask(task);
+			const data = await this.executeTask(task, controller.signal);
 			this.taskDone(taskId, data);
 		} catch (error) {
-			this.taskErrored(taskId, error);
+			clearTimeout(taskTimeout);
```

Comment:
The `clearTimeout` should probably be in the `finally` block?

Reply:
[f4a2449](https://github.com/n8n-io/n8n/pull/12101/commits/f4a2449d34877c9cf9389149fbc3f8f597a2ab17)